### PR TITLE
feat(slides): remove ion-slide, ion-slides, and IonicSwiper module

### PR DESCRIPTION
**Pull request by amandaejohnston**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

`ion-slide`, `ion-slides`, and the `IonicSwiper` module are all deprecated and slated for removal in Ionic v7.

Issue URL: N/A

## What is the new behavior?

Removes `ion-slide`, `ion-slides`, and the `IonicSwiper` module, as well as all references to them.

Some slider/swiper-related files have been kept, namely ones that are used by the `IonicSlides` module or referenced in the Swiper.js migration docs.

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

Developers using these components will need to migrate to using Swiper.js directly, optionally using the `IonicSlides` module. Guides for migration and usage:

- [Angular](https://ionicframework.com/docs/angular/slides)
- [React](https://ionicframework.com/docs/react/slides)
- [Vue](https://ionicframework.com/docs/vue/slides)

## Other information

Corresponding docs PR: https://github.com/ionic-team/ionic-docs/pull/2518

### CODE REVIEW

File: packages/vue/src/vue-component-lib/utils.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { VNode, defineComponent, getCurrentInstance, h, inject, ref, Ref } from 'vue';
 
-export interface InputProps<T> {
-  modelValue?: T;
+export interface InputProps {
```
Comment (liamdebeasi) [inline packages/vue/src/vue-component-lib/utils.ts:3]: Can we revert this change?

**Review by liamdebeasi — COMMENTED** [review]
Can we also remove: https://github.com/ionic-team/ionic-framework/blob/main/core/.stylelintrc.yml#L11

File: angular/src/index.ts [hunk 2]
```diff
@@ -50,7 +50,6 @@ export {
   createGesture,
   iosTransitionAnimation,
   mdTransitionAnimation,
-  IonicSwiper,
   IonicSlides,
```
Comment (sean-perkins) [inline angular/src/index.ts:53]: `IonicSlides` is still being exported from the Angular library entry point. Should that be removed with this? If so, should the const also be removed from `@ionic/core`?

File: core/src/index.ts [hunk 3]
```diff
@@ -27,5 +27,4 @@ export {
   popoverController,
   toastController,
 } from './utils/overlays';
-export { IonicSwiper } from './components/slides/IonicSwiper';
 export { IonicSlides } from './components/slides/IonicSlides';
```
Comment (sean-perkins) [inline core/src/index.ts:30]: This should likely be removed as well with `components/slides` directory being deleted.

File: packages/vue/src/vue-component-lib/utils.ts [hunk 4]
```diff
@@ -49,7 +49,7 @@ const getElementClasses = (ref: Ref<HTMLElement | undefined>, componentClasses:
 * @prop externalModelUpdateEvent - The external event to fire from your Vue component when modelUpdateEvent fires. This is used for ensuring that v-model references have been
 * correctly updated when a user's event callback fires.
 */
-export const defineContainer = <Props, VModelType=string|number|boolean>(
+export const defineContainer = <Props, VModelType = string | number | boolean>(
```
Comment (liamdebeasi) [inline packages/vue/src/vue-component-lib/utils.ts:52]: Any idea why this is happening?

File: BREAKING.md [hunk 5]
```diff
@@ -37,6 +38,16 @@ Developers using the component will need to migrate to a virtual scroll solution
 
 Any references to the virtual scroll types from `@ionic/core` have been removed. Please remove or replace these types: `Cell`, `VirtualNode`, `CellType`, `NodeChange`, `HeaderFn`, `ItemHeightFn`, `FooterHeightFn`, `ItemRenderFn` and `DomRenderFn`.
 
+<h4 id="version-7x-slides">Slides</h4>
+
+`ion-slides`, `ion-slide`, and the `IonicSwiper` module have been removed from Ionic.
```
Comment (sean-perkins) [inline BREAKING.md:43]: Is `IonicSwiper` a module? It appears to be a constant of shape:
```
{
  name: string;
  on: {
    afterInit(swiper: any): void;
  }
```

**Review by liamdebeasi — COMMENTED** [review]
What do you think about removing `slides.ios.vars.scss` and inlining the values for each CSS var in https://github.com/ionic-team/ionic-framework/blob/main/core/src/css/ionic-swiper.scss? Now that the slides component is removed, having the disconnect between the two files feels strange.

File: core/src/index.ts (hunk 3 above)
Comment (liamdebeasi) [reply core/src/index.ts:30]: I think `IonicSlides` is still there in the PR. This is the module that devs can use to get the same presets as the old `ion-slides`.

File: BREAKING.md (hunk 5 above)
Comment (liamdebeasi) [reply BREAKING.md:43]: IonicSwiper was a Swiper plugin before they changed how the plugins work in Swiper 6 or 7. IonicSlides is the new one.

Comment (sean-perkins) [reply BREAKING.md:43]: Should this be updated instead to:

> `ion-slides`, `ion-slide` and the `IonicSwiper` plugin have been removed from Ionic.

With module I immediately think of an Angular module being removed here.

File: packages/vue/src/vue-component-lib/utils.ts (hunk 4 above)
Comment (amandaejohnston) [reply packages/vue/src/vue-component-lib/utils.ts:52]: Probably just my editor prettifying it when I reverted the change, I'll revert that too.

**Review by liamdebeasi — APPROVED** [review]
great job!

**Review by sean-perkins — APPROVED** [review]
